src/extension/wps/web-wps/src/main/java/org/geoserver/wps/web/ProcessStatusProvider.java

```diff
@@ -65,4 +96,139 @@ public Object getPropertyValue(ExecutionStatus item) {
     }
 
+    
+
+    @Override
+    protected Filter getFilter() {
+        final String[] keywords = getKeywords();
+        Filter filter = acceptAll();
+        if (null != keywords) {
+            for (String keyword : keywords) {
+                Filter propContains = getFullSearch(keywords);
+                // chain the filters together
+                if (Filter.INCLUDE == filter) {
+                    filter = propContains;
+                } else {
+                    filter = or(filter, propContains);
+                }
+            }
+        }
+
+        return filter;
+    }
+
+    private Filter getFullSearch(String[] keywords) {
+        ProcessStatusTracker tracker = GeoServerApplication.get().getBeanOfType(
+                ProcessStatusTracker.class);
+        ProcessStatusStore store = tracker.getStore();
+        Filter ret = Filter.INCLUDE;
+        if(store.supportsPredicate()) {
+            for (String keyword : keywords) {
+                Filter propContains = Predicates.fullTextSearch(keyword);
+                // chain the filters together
+                if (Filter.INCLUDE == ret) {
+                    ret = propContains;
+                } else {
+                    ret = or(ret, propContains);
+                }
+            }
+        }else {
+            if(keywords.length>0) {
+                List<Filter> likes = new ArrayList<Filter>();
+                for(String word:keywords) {
+                    for(Property<?> prop:getProperties()) {
+                        if(prop.isSearchable()) {
+                            if(prop.equals(NODE)||
+                               prop.equals(PHASE)||
+                               prop.equals(TASK)||
+                               prop.equals(USER)) {
+                                likes.add(FF.like(FF.property(prop.getName()), "*"+word+"*"));
+                            }
+                            //TODO: support other properties
+                                
+                        }
+                    }
+                }
+                ret = FF.or(likes);
+            }
+        }
+        return ret;
+    }
+
+    @Override
+    protected List<ExecutionStatus> getFilteredItems() {
+        ProcessStatusTracker tracker = GeoServerApplication.get().getBeanOfType(
+                ProcessStatusTracker.class);
+        ProcessStatusStore store = tracker.getStore();
+        Query query = new Query("status", getFilter());
+        if(count>0) {
+            query.setStartIndex(first);
+            query.setMaxFeatures(count);
+        }
+        SortParam sort = getSort();
+        if(sort!=null) {
+            SortByImpl[] sortBys = new SortByImpl[1];
+            final Property<?> property = getProperty(sort);
+            if(property.isSearchable()) {//we really need another flag
+
+                //hard to believe there isn't an implementing class!
+                sortBys[0] = new SortByImpl(new PropertyName() {
```

Comment:
because (sadly) getName returns a String and SortByImpl demands a PropertyName (and I can't find an implementing class for that)